pkg/controller/nodeipam/ipam/cloud_cidr_allocator.go

```diff
@@ -253,43 +253,59 @@ func (ca *cloudCIDRAllocator) updateCIDRAllocation(nodeName string) error {
 		return fmt.Errorf("node %s doesn't have providerID", nodeName)
 	}
 
-	cidrs, err := ca.cloud.AliasRangesByProviderID(node.Spec.ProviderID)
+	podCIDRs, err := ca.cloud.AliasRangesByProviderID(node.Spec.ProviderID)
 	if err != nil {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
 		return fmt.Errorf("failed to allocate cidr: %v", err)
 	}
-	if len(cidrs) == 0 {
+	if len(podCIDRs) == 0 {
 		nodeutil.RecordNodeStatusChange(ca.recorder, node, "CIDRNotAvailable")
 		return fmt.Errorf("failed to allocate cidr: Node %v has no CIDRs", node.Name)
 	}
-	_, cidr, err := net.ParseCIDR(cidrs[0])
-	if err != nil {
-		return fmt.Errorf("failed to parse string '%s' as a CIDR: %v", cidrs[0], err)
+	if _, err := netutils.ParseCIDRs(podCIDRs); err != nil {
+		return err
```

Comment:
I would wrap the error so that people know it was podCIDrs that was malformed.
For completeness we should also check that node.Spec.PodCIDR and node.Spec.PodCIDRs are in the correct format.

Reply:
You should save the output of ParseCIDRs and operate on the parsed form to avoid issues with strings not being == but the actual IP specified being ==, e.g. 1::0 == 1:0:0::0